registry/storage/manifeststore.go

```diff
@@ -106,9 +106,6 @@ func (ms *manifestStore) GetByTag(tag string, options ...distribution.ManifestSe
 // content, leaving trust policies of that content up to consumers.
 func (ms *manifestStore) verifyManifest(ctx context.Context, mnfst *schema1.SignedManifest) error {
 	var errs distribution.ErrManifestVerification
-	if mnfst.Name != ms.repository.Name() {
```

Comment:
I think we still need to validate the name, but we don't need to match it to the repo.